packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -201,40 +217,47 @@ export function makeRumPublicApi(
     }),
 
     setUser: monitor((newUser: User) => {
-      const sanitizedUser = sanitizeUser(newUser)
-      if (sanitizedUser) {
-        user = sanitizedUser
-      } else {
+      if (typeof newUser !== 'object' || !newUser) {
         display.error('Unsupported user:', newUser)
+      } else {
+        userContextManager.setContext(sanitizeUser(newUser as Context))
       }
     }),
 
+    getUser: monitor(userContextManager.getContext),
+
+    setUserProperty: monitor(userContextManager.setContextProperty),
+
+    removeUserProperty: monitor(userContextManager.removeContextProperty),
+
+    clearUser: monitor(userContextManager.clearContext),
+
     removeUser: monitor(() => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏add a `@deprecated` comment above the `removeUser` function